File: api/app/schemas/operations.py
from pydantic import BaseModel


class OperationsOverview(BaseModel):
    total_assets: int
    published_assets: int
    reviewing_assets: int
    low_quality_assets: int
    pending_access_requests: int


class OperationsTask(BaseModel):
    asset_id: str
    slug: str
    title: str
    status: str
    reason: str
    priority: str
    target_url: str


class OperationsTasks(BaseModel):
    items: list[OperationsTask]
    total: int


class RecentActivity(BaseModel):
    id: str
    asset_id: str
    asset_title: str
    action: str
    from_status: str | None
    to_status: str
    reason: str
    created_at: str


class RecentActivities(BaseModel):
    items: list[RecentActivity]
    total: int

File: web/src/types/operations.ts
export type OperationsOverview = {
  total_assets: number;
  published_assets: number;
  reviewing_assets: number;
  low_quality_assets: number;
  pending_access_requests: number;
};

export type OperationsTask = {
  asset_id: string;
  slug: string;
  title: string;
  status: string;
  reason: string;
  priority: "high" | "medium" | "low";
  target_url: string;
};

export type OperationsTasks = {
  items: OperationsTask[];
  total: number;
};

export type RecentActivity = {
  id: string;
  asset_id: string;
  asset_title: string;
  action: string;
  from_status: string | null;
  to_status: string;
  reason: string;
  created_at: string;
};

export type RecentActivities = {
  items: RecentActivity[];
  total: number;
};
